src/Mappers/Root/RootTypeMapperInterface.php

```diff
@@ -27,19 +27,17 @@ interface RootTypeMapperInterface
 {
     /**
      * @param (OutputType&GraphQLType)|null $subType
-     * @param ReflectionMethod|ReflectionProperty $reflector
      *
      * @return OutputType&GraphQLType
```

Comment:
@bladl was reviewing this further and prepping a release update and noticed inconsistencies on the return typing for this interface.  Was this an oversight, or can you elaborate further here?  Also, the PHPDoc for `$subType` isn't consistent and should probably just be removed right?  Also the return typehint doc?

Reply:
I researched existing codebase before making changes. My opinion was that intersection types `OutputType&GraphQLType` and `InputType&GraphQLType` is important here and we should leave PHPDoc as is. 
As I think its not inconsistencies, its clarification. 

We could use [pure intersection types](https://www.php.net/releases/8.1/en.php#pure_intersection_types)  , but they are supported only by PHP 8.1

This PHPDoc with intersection type was added in 2019, and I think we should keep it as is until minimum supported PHP version will be 8.1

Reply:
@bladl I see what you're saying, I guess I should clarify some of my questions:

For `toGraphQLInputType` the return type is a union of `InputType|GraphQLType`.  However, `toGraphQLOutputType` the return type is only `OutputType`.  Does the `toGraphQLInputType` support a return type that's a `GraphQLType` and not an `InputType`?

As for the `$subType` parameter, it's similar situation with `toGraphQLInputType` supporting a union and `toGraphQLOutputType only supporting `?OutputType`.

I totally see what you mean about the intersection type and agree with having that PHPDoc.  That clarifies that bit.  Does `toGraphQLInputType` just need to respect the same as `toGraphQLOutputType`?  It seems the typing for `toGraphQLOutputType` is much better.  Where `toGraphQLInputType` seems inaccurate or unnecessarily loose.

Reply:
Now I see. You are right!
I will open a PR with fixes soon.

Seems like usage of union type was some kind of workaround for intersection type.

Reply:
@oojacoboo Please, review the changes #520

Reply:
@bladl awesome thanks!